middleware/retry.go

```diff
@@ -0,0 +1,82 @@
+package middleware
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/livekit/psrpc"
+)
+
+type RetryOptions struct {
+	MaxAttempts int
+	Timeout     time.Duration
+	Backoff     time.Duration
+}
+
```

Comment:
This seems to use a constant backoff, with no option to have an exponential one. What is the use case here? Would it make any sense to honor context cancellation/timeout here as well?

Reply:
if we need something more sophisticated we can add something like `BackoffFunc func(attempt int) time.Duration` later. my use case for this was resolving races between remote node subscriptions and rpcs where this simple strategy is good enough.

ending retries on parent cancellation is a good call. added below.